### PR TITLE
#11912: Postgresql negative default values support for integer/float/decimal columns

### PR DESCRIPTION
**Pull request by nsknewbie**
| Q | A |
| --- | --- |
| Is bugfix? | yes |
| New feature? | no |
| Breaks BC? | possibly yes, if users expected `0` instead of correct default value |
| Tests pass? | yes |
| Fixed issues | #11912 |

New regexp looks bulky, it can be improved for sure, but i tried to make it stricter

### CODE REVIEW

**Comment by cebe** [comment]
thanks! the tests could possibly be added for other dbms too so we can check whether it needs to be fixed in other places too.

**Comment by nsknewbie** [comment]
@cebe added more tests, seems all's fine

File: framework/db/pgsql/Schema.php [hunk 1]
```diff
@@ -468,7 +468,7 @@ protected function findColumns($table)
                     $column->defaultValue = bindec(trim($column->defaultValue, 'B\''));
                 } elseif (preg_match("/^'(.*?)'::/", $column->defaultValue, $matches)) {
                     $column->defaultValue = $matches[1];
-                } elseif (preg_match('/^(.*?)::/', $column->defaultValue, $matches)) {
+                } elseif (preg_match('/^(?:\()?(.*?)(?(1)\))(?:::.+)?$/', $column->defaultValue, $matches)) {
```
Comment (cebe) [inline framework/db/pgsql/Schema.php]: just for me to fully understand this: why did you add the part that requires text after `::`? are you sure that can not be empty?

**Comment by cebe** [comment]
please also add a line to the CHANGLOG.

File: framework/db/pgsql/Schema.php (hunk 1 above)
Comment (nsknewbie) [inline framework/db/pgsql/Schema.php]: there should be a data type after `::`

**Comment by cebe** [comment]
Merged, thank you!
